src/app/collective-rewards/rewards/backers/hooks/useGetRBI.ts

```diff
@@ -0,0 +1,81 @@
+import { Address } from 'viem'
+import { usePricesContext } from '@/shared/context/PricesContext'
+import Big from '@/lib/big'
+import { useMemo } from 'react'
+import { WeiPerEther } from 'ethers'
+import { useCycleContext } from '@/app/collective-rewards/metrics'
+import { useBackerRewardsContext, useGetBackerStakingHistory, Token } from '@/app/collective-rewards/rewards'
+
+const useGetTokenRewards = ({ address, symbol }: Token) => {
+  const { prices } = usePricesContext()
+  const { data, isLoading, error } = useBackerRewardsContext()
+  const price = prices[symbol]?.price ?? 0
+
+  const { earned, claimed } = data[address]
+  const totalEarned = Object.values(earned).reduce((acc, earned) => acc + earned, 0n)
+  const totalClaimed = Object.values(claimed).reduce(
+    (acc, value) => acc + value.reduce((acc, value) => acc + value.args.amount_, 0n),
+    0n,
+  )
+
+  return {
+    data: Big((totalClaimed + totalEarned).toString())
+      .div(WeiPerEther.toString())
+      .mul(price),
+    isLoading,
+    error,
+  }
+}
+
+export const useGetBackerRBI = (backer: Address, { rbtc, rif }: Record<string, Token>) => {
+  const {
+    data: stakingHistory,
+    isLoading: stakingHistoryLoading,
+    error: stakingHistoryError,
+  } = useGetBackerStakingHistory(backer)
+  const {
+    data: rbtcRewards,
+    isLoading: rbtcRewardsLoading,
+    error: rbtcRewardsError,
+  } = useGetTokenRewards(rbtc)
+  const { data: rifRewards, isLoading: rifRewardsLoading, error: rifRewardsError } = useGetTokenRewards(rif)
```

Comment:
Wouldn't make more sense to multiply by the price outside so we can call the `useGetTokenRewards` only once?

Reply:
Yeah we could do it, but the `useGetTokenRewards` would be call twice since we need the rewards for RBTC and RIF, isnt it?

Reply:
You're right, it doesn't make sense to split it 👍🏼